pilot/pkg/model/envoyfilter.go

```diff
@@ -163,11 +164,18 @@ func proxyMatch(proxy *Proxy, cp *EnvoyFilterConfigPatchWrapper) bool {
 	return true
 }
 
-func (efw *EnvoyFilterWrapper) Key() string {
+// Returns all the envoyfilters that wrapped.
+func (efw *EnvoyFilterWrapper) Keys() []string {
```

Comment:
can Keys be constructed as part of EnvoyFilterWrapper creation? instead of computing them whenever it is called?

Reply:
This can save some cpu cycles, but I think it is not very good to cache any attributes that can be inferred from the existing fields.

Reply:
I think it depends on how frequently you call and what is being cached. Since there wont be may patches and patches wont change after wrapper is constructed - it feels safe to cache it as there wont be any memory pressure. But up to you

Reply:
Let me rethink about it.

Reply:
@ramaraochavali Cache the keys won't help here. 
As for cds, we generate a envoyfilterwrapper for each proxy once.  https://github.com/istio/istio/blob/888434e9c6f7817fa4d004ce450fcb0069365f4f/pilot/pkg/networking/core/v1alpha3/cluster.go#L79